core/src/muxing/one_substream.rs

```diff
@@ -0,0 +1,137 @@
+// Copyright 2019 Parity Technologies (UK) Ltd.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a
+// copy of this software and associated documentation files (the "Software"),
+// to deal in the Software without restriction, including without limitation
+// the rights to use, copy, modify, merge, publish, distribute, sublicense,
+// and/or sell copies of the Software, and to permit persons to whom the
+// Software is furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+// OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+// DEALINGS IN THE SOFTWARE.
+
+use crate::{Endpoint, muxing::StreamMuxer};
+use futures::prelude::*;
+use parking_lot::Mutex;
+use std::{io, sync::atomic::{AtomicBool, Ordering}};
+use tokio_io::{AsyncRead, AsyncWrite};
+
+/// Implementation of `StreamMuxer` that allows only one substream on top of a connection,
+/// yielding the connection itself.
+///
+/// Nothing happens on the wire.
```

Comment:
That seems a bit misleading. Surely it only means that there is no substream protocol used on the wire?

Reply:
Yes. I don't know how to phrase that.

Reply:
How about `No substream protocol used on the wire.`?

Reply:
I changed it in the latest commit. Let me know.

Reply:
I think you forgot to check-in the new file?

Reply:
Oops